src/ucs/sys/sock.c

```diff
@@ -144,6 +144,20 @@ ucs_status_t ucs_socket_setopt(int fd, int level, int optname,
     return UCS_OK;
 }
 
+ucs_status_t ucs_socket_getopt(int fd, int level, int optname,
+                               void *optval, socklen_t* optlen_ptr)
```

Comment:
but it's an [out] param

Reply:
make it [in]

Reply:
but need to return the caller the number of bytes actually stored by getsockopt() in optval.

Reply:
why? where is it used?

Reply:
the current caller of ucs_socket_getopt in tcp_sockcm doesnt use this value but it is set by getsockopt so why not return it if possible? imho we should make the usage of ucs_socket_getopt generic and if getsockopt provides this value, why ignore it?

Reply:
we can add if needed, for now removing it will make the code simpler

Reply:
ok will remove.
i disagree though, we are adding a wrapper for getsockopt() and therefore, imho, should keep its properties. the caller of the function shouldn't affect the implementation of this since it can choose what to do with the returned values.

Reply:
IMO the only thing the caller can do is print an error if the returned length is not same as input length, and this check we can do inside the wrapper